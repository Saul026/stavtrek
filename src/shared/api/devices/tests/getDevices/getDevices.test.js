const getDevices = require('./getDevices');
const axios = require('axios')

jest.mock('axios')

describe('getData', () => {
    let response
    beforeEach(() => {
        response = {
            data: [
                {
                    "id": 9,
                    "attributes": {
                      "deviceImage": "device.webp",
                      "processing.copyAttributes": "",
                      "decoder.timezone": "",
                      "speedLimit": 53.9956803455724,
                      "web.reportColor": "красный",
                      "deviceInactivityStart": 15
                    },
                    "groupId": 0,
                    "calendarId": 0,
                    "protocol": null,
                    "name": "Монтажник тест ",
                    "uniqueId": "135079",
                    "status": "online",
                    "lastUpdate": "2024-08-08T17:19:49.286+00:00",
                    "positionId": 19562950,
                    "phone": null,
                    "model": null,
                    "contact": null,
                    "category": null,
                    "disabled": false,
                    "expirationTime": null
                  },
                  {
                    "id": 2,
                    "attributes": {
                
                    },
                    "groupId": 0,
                    "calendarId": 0,
                    "protocol": null,
                    "name": "Тест",
                    "uniqueId": "550393",
                    "status": "offline",
                    "lastUpdate": "2023-11-27T07:47:46.000+00:00",
                    "positionId": 131758,
                    "phone": null,
                    "model": null,
                    "contact": null,
                    "category": null,
                    "disabled": false,
                    "expirationTime": null
                  }
            ]
        }
    })
    test('Корректное значение', async () => {
        axios.get.mockReturnValue(response);
        const data = await getDevices()
        expect(data).toEqual(response)
        expect(data).toMatchSnapshot()
    })
})
