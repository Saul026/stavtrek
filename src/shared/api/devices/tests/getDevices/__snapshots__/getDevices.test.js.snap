// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getData Корректное значение 1`] = `
{
  "data": [
    {
      "attributes": {
        "decoder.timezone": "",
        "deviceImage": "device.webp",
        "deviceInactivityStart": 15,
        "processing.copyAttributes": "",
        "speedLimit": 53.9956803455724,
        "web.reportColor": "красный",
      },
      "calendarId": 0,
      "category": null,
      "contact": null,
      "disabled": false,
      "expirationTime": null,
      "groupId": 0,
      "id": 9,
      "lastUpdate": "2024-08-08T17:19:49.286+00:00",
      "model": null,
      "name": "Монтажник тест ",
      "phone": null,
      "positionId": 19562950,
      "protocol": null,
      "status": "online",
      "uniqueId": "135079",
    },
    {
      "attributes": {},
      "calendarId": 0,
      "category": null,
      "contact": null,
      "disabled": false,
      "expirationTime": null,
      "groupId": 0,
      "id": 2,
      "lastUpdate": "2023-11-27T07:47:46.000+00:00",
      "model": null,
      "name": "Тест",
      "phone": null,
      "positionId": 131758,
      "protocol": null,
      "status": "offline",
      "uniqueId": "550393",
    },
  ],
}
`;
